src/main/groovy/wooga/gradle/snyk/SnykPlugin.groovy

```diff
@@ -66,16 +80,143 @@ class SnykPlugin implements Plugin<Project> {
             mapExtensionPropertiesToMonitorTask(it, extension)
         }
         // Register an install task (to be used to install the snyk binary if need be)
-        registerSnykTasks(project, extension)
+        registerSnykInstall(project, extension, snykInstall)
+        registerTaskGroupAndDescription(project)
+        hookStrategyLifecycleTasks(project, extension)
+    }
+
+    @Override
+    SnykPluginExtension registerProject(Project project, SnykRootPluginExtension parentExtension) {
+        def projectName = project.rootProject.name + project.path
+        def snykTest = project.tasks.register(TEST_TASK_NAME, Test)
+        def snykReport = project.tasks.register(REPORT_TASK_NAME, Report)
+        def snykMonitor = project.tasks.register(MONITOR_TASK_NAME, Monitor)
+
+        def extension = project.extensions.create(SnykPluginExtension, EXTENSION_NAME, DefaultSnykPluginExtension, project, snykTest, snykMonitor, snykReport)
+
+        setupProjectRegistration(project, projectName, extension, parentExtension, snykTest, snykMonitor, snykReport)
+        registerTaskGroupAndDescription(project)
+        extension.packageFile.set(null)
+        extension.workingDirectory.convention(parentExtension.workingDirectory)
+        extension.subProject.set(project.name)
+        extension
+    }
+
+    SnykPluginExtension registerProject(File projectFile, SnykRootPluginExtension parentExtension) {
+        def relativeProjectFile = project.relativePath(projectFile)
+        def projectName = project.name + ":" + relativeProjectFile
+
+        if (projectFile.isFile()) {
+            logger.info("register snyk project file: ${projectFile.absolutePath}")
+        } else {
+            logger.info("register snyk project at path: ${projectFile.absolutePath}")
+        }
```

Comment:
Should it not check if the given project file does not exist?

Reply:
Yes correct. That as well. I saw already during testing that `isFile` will be false if the file does not exist. I will add the check and a test.

Reply:
Got it!